### PR TITLE
Modify bealign wrapper to support selection analyses

### PR DESCRIPTION
**Pull request by davebx**
- Allow choosing a background to also align against the reference.
- Add data manager for selection backgrounds.
- Save reference used when aligning against a built-in CoV2 or HIV gene.
- Use gawk instead of awk for OSX compatibility.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR updates an existing tool or tool collection

### CODE REVIEW

File: tools/bioext/.shed.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ remote_repository_url: https://github.com/davebx/bioext-gx/
 homepage_url: https://pypi.python.org/pypi/biopython-extensions/
 type: unrestricted
 categories:
- - Next Gen Mappers
+ - Tools
```
Comment (bernt-matthias) [inline tools/bioext/.shed.yml:8]: Is this a valid category?

File: tools/bioext/bealign.xml [hunk 2]
```diff
@@ -138,6 +184,15 @@
             <param name="reference" ftype="fasta" value="bealign-in-ref-2.fa" />
             <output name="output" file="bealign-out3.bam" ftype="bam" lines_diff="2"/>
         </test>
+        <test>
+            <param name="input" ftype="fasta" value="bealign-in3.fa" />
```
Comment (bernt-matthias) [inline tools/bioext/bealign.xml:188]: Can you add a test for the new conditional?

File: tools/bioext/.shed.yml (hunk 1 above)
Comment (davebx) [reply tools/bioext/.shed.yml:8]: No, that was something I forgot to remove before committing.

File: tools/bioext/bealign.xml [hunk 3]
```diff
@@ -16,17 +18,40 @@
...
+            --expected-identity $advanced.expected_identity
+        #end if
+        --keep-reference
+        --score-matrix $advanced.score_matrix
+        $advanced.reverse_complement 
+        background.fa background.bam
+    #end if
+    #set $reference_name = str($select_reference.reference)
+    #if $select_reference.reference_type == 'preset' and $select_reference.save_reference:
+        && cp `python $__tool_directory__/bealign_ref.py $select_reference.reference` reference.fa
```
Comment (bernt-matthias) [inline tools/bioext/bealign.xml:53]: This looks odd. Maybe:

```suggestion
        && cp `$select_reference.reference` `$saved_reference`
```

File: tools/bioext/bealign.xml [hunk 4]
```diff
@@ -110,7 +156,13 @@
         </section>
     </inputs>
     <outputs>
-        <data name="output" format="bam" from_work_dir="alignment.bam" />
+        <data name="output" format="bam" from_work_dir="alignment.bam" label="${tool.name} on ${on_string} - Aligned Sequences" />
+        <data name="background" format="bam" from_work_dir="background.bam" label="${tool.name} on ${on_string} - Background" >
+            <filter>clade_bg['add_background']</filter>
+        </data>
+        <data name="saved_reference" format="fasta" from_work_dir="reference.fa" label="${tool.name} on ${on_string} - Reference" >
```
Comment (bernt-matthias) [inline tools/bioext/bealign.xml:163]:
```suggestion
        <data name="saved_reference" format="fasta" label="${tool.name} on ${on_string} - Reference">
```

File: tools/bioext/bealign.xml (hunk 3 above)
Comment (bernt-matthias) [reply tools/bioext/bealign.xml:53]: Ah, I see. Then its fine.

File: tools/bioext/bealign.xml [hunk 5]
```diff
@@ -138,6 +190,31 @@
             <param name="reference" ftype="fasta" value="bealign-in-ref-2.fa" />
             <output name="output" file="bealign-out3.bam" ftype="bam" lines_diff="2"/>
         </test>
+        <test>
```
Comment (bernt-matthias) [inline tools/bioext/bealign.xml:193]: `expect_num_outputs` would be good to add here.

**Review by bernt-matthias — COMMENTED** [review]
`test-data/bealign-ref-out1.fa ` is empty. Is this expected?

File: tools/bioext/bealign.xml [hunk 6]
```diff
@@ -16,17 +18,40 @@
...
+            --expected-identity $advanced.expected_identity
+        #end if
+        --keep-reference
+        --score-matrix $advanced.score_matrix
+        $advanced.reverse_complement 
+        background.fa '$background'
+    #end if
+    #set $reference_name = str($select_reference.reference)
+    #if $select_reference.reference_type == 'preset' and $select_reference.save_reference:
+        && cp `python $__tool_directory__/bealign_ref.py $select_reference.reference` '$saved_reference'
```
Comment (bernt-matthias) [inline tools/bioext/bealign.xml:53]: Quote __tool_directory__

File: tools/bioext/bealign.xml [hunk 7]
```diff
@@ -81,13 +106,34 @@
                     <option value="CoV2-ORF10">SARS-CoV-2: ORF10</option>
                     <option value="CoV2-RdRp">SARS-CoV-2: RNA-dependent RNA polymerase</option>
                 </param>
+                <param name="save_reference" type="boolean" display="radio" label="Save this reference to your history" />
             </when>
             <when value="dataset">
                 <param argument="--reference" type="data" format="fasta" label="Reference sequences" />
             </when>
         </conditional>
+        <conditional name="clade_bg">
```
Comment (bernt-matthias) [inline tools/bioext/bealign.xml:115]: Instead of the conditional on could make the data or select parameter optional.

**Review by bernt-matthias — COMMENTED** [review]
Still some of the Test data are empty, eg tools/bioext/test-data/bealign-out1.bam

File: tools/bioext/bealign.xml (hunk 7 above)
Comment (davebx) [reply tools/bioext/bealign.xml:115]: I think it should be cleaner now.

**Comment by davebx** [comment]
@bernt-matthias I think the test data should be sorted now, I noticed that some tests were redundant and combined them.

**Review by bgruening — COMMENTED** [review]
* data_managers/data_manager_selection_background/README -> should that be removed?

**Comment by davebx** [comment]
Thanks, @bgruening. I had intended to either delete that file or remove it, but forgot.

**Review by bgruening — APPROVED** [review]

**Pull request merged**